infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -290,21 +304,84 @@ private ServerContext getServerContext(Map<String, Object> opts) {
     }
 
     /**
-     * Start Corfu Log Replication Discovery Service
+     * Start Corfu Log Replication Discovery Service after the local node version matches the cluster version, i.e.,
+     * any ongoing rolling upgrade completes.
      *
      * @param serverContext server context (server information)
      */
     private void startDiscoveryService(ServerContext serverContext) {
 
+        CorfuRuntime runtime = getRuntime(serverContext);
+        CorfuStore corfuStore = new CorfuStore(runtime);
+
+        LogReplicationUpgradeManager upgradeManager =
+                new LogReplicationUpgradeManager(corfuStore, serverContext.getPluginConfigFilePath());
+
+        // Check if an upgrade is in progress.  If it is, wait for it to complete
+        log.info("Wait for any ongoing rolling upgrade to complete.");
+
+        while(isUpgradeInProgress(upgradeManager, corfuStore)) {
```

Comment:
it can be stuck of take a long time in such a scenario.  But we need to block until it has fully completed and start once we can.  So we cannot have a timeout in this case.